src/app/components/Ionic/IonicProvider.tsx

```diff
@@ -0,0 +1,43 @@
+import { createContext, FC, PropsWithChildren, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Capacitor, type PluginListenerHandle } from '@capacitor/core'
+import { App } from '@capacitor/app'
+
+const IonicContext = createContext<undefined>(undefined)
+
+let backButtonListenerHandle: PluginListenerHandle | undefined
+
+const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    /**
+     * The back button refers to the physical back button on an Android device and should not be confused
+     * with either the browser back button or ion-back-button.
+     */
+    backButtonListenerHandle = App.addListener('backButton', ({ canGoBack }) => {
+      if (!canGoBack) {
+        App.exitApp()
+      } else {
+        navigate(-1)
+      }
+    })
+
+    return () => {
+      if (backButtonListenerHandle) {
+        backButtonListenerHandle.remove()
+        backButtonListenerHandle = undefined
+      }
+    }
```

Comment:
Is this different from
```tsx
    const backButtonListenerHandle = App.addListener('backButton', ({ canGoBack }) => {
      if (!canGoBack) {
        App.exitApp()
      } else {
        navigate(-1)
      }
    })

    return () => {
      backButtonListenerHandle.remove()
    }
```

Reply:
It's not. I wanted to do it because of StrictMode(running re-render x2 in dev mode), but I should have skipped attaching the listener, if it is already defined.
Anyway inline const is better. Will update with follow up PR.

Reply:
Fixed in https://github.com/oasisprotocol/oasis-wallet-web/pull/1932